File: src/main/kotlin/ympa_kotlin_wiremock_server/models/FeedbackDTO.kt
@file:Suppress(
    "RemoveRedundantQualifierName",
    "unused",
)

package ympa_kotlin_wiremock_server.models

import com.fasterxml.jackson.annotation.JsonProperty

data class FeedbackDTO(
    @field:JsonProperty("comments")
    val comments: kotlin.collections.List<FeedbackCommentDTO>,

    @field:JsonProperty("id")
    val id: kotlin.Long? = null,

    @field:JsonProperty("createdAt")
    val createdAt: java.time.OffsetDateTime? = null,

    @field:JsonProperty("text")
    val text: kotlin.String? = null,

    @field:JsonProperty("state")
    val state: FeedbackStateType? = null,

    @field:JsonProperty("author")
    val author: FeedbackAuthorDTO? = null,

    @field:JsonProperty("pro")
    val pro: kotlin.String? = null,

    @field:JsonProperty("contra")
    val contra: kotlin.String? = null,

    @field:JsonProperty("shop")
    val shop: FeedbackShopDTO? = null,

    @field:JsonProperty("resolved")
    val resolved: kotlin.Boolean? = null,

    @field:JsonProperty("verified")
    val verified: kotlin.Boolean? = null,

    @field:JsonProperty("recommend")
    val recommend: kotlin.Boolean? = null,

    @field:JsonProperty("grades")
    val grades: FeedbackGradesDTO? = null,

    @field:JsonProperty("order")
    val order: FeedbackOrderDTO? = null,

)

File: ympa_fsharp_giraffe_server/src/model/FeedbackDTO.fs
namespace ympa_fsharp_giraffe_server.Model

open System
open System.Collections.Generic
open ympa_fsharp_giraffe_server.Model.FeedbackAuthorDTO
open ympa_fsharp_giraffe_server.Model.FeedbackCommentDTO
open ympa_fsharp_giraffe_server.Model.FeedbackGradesDTO
open ympa_fsharp_giraffe_server.Model.FeedbackOrderDTO
open ympa_fsharp_giraffe_server.Model.FeedbackShopDTO
open ympa_fsharp_giraffe_server.Model.FeedbackStateType

module FeedbackDTO =

  //#region FeedbackDTO


  type FeedbackDTO = {
    Id : int64;
    CreatedAt : Nullable<DateTime>;
    Text : string;
    State : FeedbackStateType;
    Author : FeedbackAuthorDTO;
    Pro : string;
    Contra : string;
    Comments : FeedbackCommentDTO[];
    Shop : FeedbackShopDTO;
    Resolved : bool;
    Verified : bool;
    Recommend : bool;
    Grades : FeedbackGradesDTO;
    Order : FeedbackOrderDTO;
  }
  //#endregion
